cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -618,12 +630,40 @@ public Secret generateBrokersSecret() {
         return portList;
     }
 
-    private List<Volume> getVolumes(boolean isOpenShift) {
-        List<Volume> volumeList = new ArrayList<>();
+    /**
+     * Fill the with volumes, persistent volume claims and related volume mount paths for the storage
+     * It's called recursively on the related inner volumes if the storage is of {@link Storage#TYPE_JBOD} type
+     *
+     * @param storage the Storage instance from which building volumes, persistent volume claims and related volume mount paths
+     */
+    private void setDataVolumesClaimsAndMountPaths(Storage storage) {
+        Integer id = null;
         if (storage instanceof EphemeralStorage) {
-            volumeList.add(createEmptyDirVolume(VOLUME_NAME));
+            id = ((EphemeralStorage) storage).getId();
+        } else if (storage instanceof PersistentClaimStorage) {
+            id = ((PersistentClaimStorage) storage).getId();
+        } else if (storage instanceof JbodStorage) {
+            for (SingleVolumeStorage volume: ((JbodStorage) storage).getVolumes()) {
+                setDataVolumesClaimsAndMountPaths(volume);
+            }
+            return;
         }
```

Comment:
Because I'm a paranoid programmer I would add an `else { throw }` after each of the supported cases.